WPMedia/fusion-news-theme-blocks
# PEN-1493: [CMG] Play in Place – Promo Blocks

## Description
Added 'Video Promo - Play in place' functionality to 'large-promo-block', 'extra-large-promo-block', 'top-table-list-block' (LG & XL items)

## Jira Ticket
- [PEN-1493](https://arcpublishing.atlassian.net/browse/PEN-1493)

## Acceptance Criteria
1. PageBuilder users can enable a new boolean Custom Field called Play video in place for the following: 
a. Extra Large Promo Block
i. Should be under the Art custom field group
b. Large Promo Block
i. Should be under the Art custom field group
c. Top Table List – XL and Large display sizes
i. Should be under the Extra large story settings and Large story settings custom field groups respectively
2. The default setting for Play video in place should be false 
3. If Play video in place is enabled, then if the content item being rendered is either type: video or a type:story with a type:video as lead_art, the promo should use that video embed as the promo art (instead of the basic image) 
a. If a reader presses Play on one of these videos, the video starts playing in place (as opposed to taking the reader to the video or article page) 
b. The Image ratio custom field is not expected to apply to the video (rather, the video embed will display at whatever its natural ratio is, how it was uploaded to Video Center)
c. The labels implemented in  are not expected to display on play in place videos (rather, there will be a play button visible from the video embed itself) 
4. If the user clicks the headline, they should go to that content’s page (as happens now – no change to functionality is needed here) 
5. Tests and documentation are updated to cover this new functionality

## Test Steps
1. In `Fusion-News-Theme` repo, checkout `master` & `git pull`
2. After checking out this feature branch in `fusion-news-theme-blocks`, run `rm -rf ./node_modules && npx lerna clean && npm i && npx lerna bootstrap && cd blocks/header-nav-chain-block && npm i && cd ../..`
3. Run `npx fusion start -f -l @wpmedia/top-table-list-block,@wpmedia/extra-large-promo-block,@wpmedia/large-promo-block,@wpmedia/video-player-block` in `Fusion-News-Theme`
4. Open a test page in PB editor and add one of each of the following blocks to the **main** section: `Large Promo Block`, `Extra Large Promo Block`, `Top Table List Block`
5. Configure them with the following setups:
![image](https://user-images.githubusercontent.com/26662906/100116295-2ee97f00-2e39-11eb-8bdd-e860c90a6eba.png)
![image](https://user-images.githubusercontent.com/26662906/100116175-0b263900-2e39-11eb-9883-84ef09776e4c.png)
![image](https://user-images.githubusercontent.com/26662906/100116899-e088b000-2e39-11eb-8ce3-2b08e8fbde86.png)
6. Stage, publish, open your test page in new tab
7. You should see videos loading for _most_ of the resulting promo items returned from the story feed query that _have_ a video available in `promo_items.lead_art`. A couple of the videos appear to be broken for the feed I'm getting but this appears to be an issue with something setup on the Video Center side of things. The broken instances look like this but we can ignore them since it doesn't appear to be an issue that has anything to do with the blocks code:
![image](https://user-images.githubusercontent.com/26662906/100117456-6c024100-2e3a-11eb-981e-291aac75187c.png)
8. Make sure that all of the working video players (ones that appear to be loading as expected with background image and play button) work properly when you click "play". They should also all work simultaneously. Finally make sure everything looks good style-wise (no obvious regressions) and renders as expected across breakpoints.

## Effect Of Changes
### Before
Large and Extra Large Promo blocks that receive video data through `lead_art` were unable to actually render the video player on the promo card. They would instead render the promo image pulled from the video data.

### After
Now, Large and Extra Large Promo Blocks (as well as the ones existing in Top Table List Block) are able to render an actual video player so the user can play the video directly from the promo card (versus following a link to the page with a video on it). This is controlled through the `Play Video In Place` custom field. When "unchecked", the promo item(s) will continue using the video thumbnail image instead of the actual video player.

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
